src/treebuilders/DerivativeCalculator.cpp

```diff
@@ -110,6 +110,9 @@ void DerivativeCalculator<D>::calcNode(MWNode<D> &gNode) {
             }
         }
     }
+    // Multiply appropriate scaling factor
+    const double sf = gNode.getMWTree().getMRA().getWorldBox().getScalingFactor()[this->applyDir];
+    for (int i = 0; i < gNode.getNCoefs(); i++) gNode.getCoefs()[i] *= 1.0/sf;
```

Comment:
I think it looks like a good idea! Currently, coefs is a pointer, this should probably be changed to an std::vector, then
your second for_each will work. @stigrj  what's your opinion on this?

Reply:
The MW coefficients is *the* performance critical point in the code, and it's an array of ~10^4 doubles, so I can't imagine it'd be a good idea to make it an `std::vector`.

Reply:
Without looking at the generated code it's hard to say which approach will yield faster code. On access to the data, raw and STL should be the same. Unless you're accessing `std::vector` with `at` (which bounds check), that is. Memory-wise, I agree, `std::vector` will cost more, but again, that should be checked. If you `reserve` space beforehand even that cost should be easily amortized. Note that returning iterators can as well work while storing a raw pointer in the class.

Reply:
Yes, as long as the raw data is stored consecutively in memory and we don't lose vectorization it's probably fine. I guess there's no problem if you `reserve` space as you mention? However, there's some nasty stuff underneath in the memory management of the nodes and coefficients, so it will not be a simple transition to make. Returning an iterator might be a viable option, though.